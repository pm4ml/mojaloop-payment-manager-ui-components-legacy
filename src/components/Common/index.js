import './Common.scss';
import '../../icons/mule/check-small.svg';
import '../../icons/mule/close-small.svg';

import PropTypes from 'prop-types';
import React from 'react';

import * as utils from '../../utils/common';
import Button from '../Button';
import Icon, { iconSizes } from '../Icon';
import Row from '../Row';
import Spinner from '../Spinner';
import Tooltip from '../Tooltip';

const Loader = ({ size }) => (
  <div className="mb-input__inner-icon mb-loader">
    <Spinner size={iconSizes[size]} />
  </div>
);

Loader.propTypes = {
  size: PropTypes.oneOf(['s', 'm', 'l']),
};
Loader.defaultProps = {
  size: 'l',
};

const Placeholder = ({ label, size, active }) => {
  // The Placeholder that renders inside an input

  const placeholderClassName = utils.composeClassNames([
    'mb-input__placeholder',
    size === 's' && 'mb-input__placeholder--small',
    size === 'm' && 'mb-input__placeholder--medium',
    size === 'l' && 'mb-input__placeholder--large',
    size === 's' && active && 'mb-input__placeholder--active-small',
    size === 'm' && active && 'mb-input__placeholder--active-medium',
    size === 'l' && active && 'mb-input__placeholder--active-large',
  ]);

  return <label className={placeholderClassName}>{label}</label>;
};

Placeholder.propTypes = {
  label: PropTypes.string,
  active: PropTypes.bool,
  size: PropTypes.oneOf(['s', 'm', 'l']),
};

Placeholder.defaultProps = {
  label: undefined,
  active: false,
  size: 'l',
};

const InnerButton = ({ className, size, kind, onClick, label, disabled, noFill, icon }) => {
  // Internal button used by inputs

  let innerButtonSize;
  if (size === 'l') {
    innerButtonSize = 'm';
  } else if (size === 'm') {
    innerButtonSize = 's';
  } else {
    innerButtonSize = 'xs';
  }

  const innerButtonClassName = utils.composeClassNames([
    className,
    'mb-input__inner-button',
    size === 's' && 'mb-input__inner-button--small',
    size === 'm' && 'mb-input__inner-button--medium',
    size === 'l' && 'mb-input__inner-button--large',
  ]);
  return (
    <Button
      kind={kind}
      size={innerButtonSize}
      className={innerButtonClassName}
      icon={icon}
      noFill={noFill}
      onClick={onClick}
      tabIndex="-1"
      label={label}
      disabled={disabled}
    />
  );
};

InnerButton.propTypes = {
  className: PropTypes.string,
  size: PropTypes.oneOf(['s', 'm', 'l']),
  kind: PropTypes.string,
  icon: PropTypes.string,
  onClick: PropTypes.func,
  label: PropTypes.string,
  disabled: PropTypes.bool,
};

InnerButton.defaultProps = {
  className: undefined,
  size: 'l',
  kind: 'primary',
  icon: undefined,
  onClick: undefined,
  label: undefined,
  disabled: false,
};

const ActiveValidationIcon = () => (
  <Icon className="validation__message__icon--active" name="close-small" size={12} />
);
const InactiveValidationIcon = () => (
  <Icon className="validation__message__icon--inactive" name="check-small" size={14} />
);
const UnsetValidationIcon = () => <div className="validation__message__icon--unset" />;

const ValidationIcon = ({ active }) => {
  if (active === true) {
    return <ActiveValidationIcon />;
  } else if (active === false) {
    return <InactiveValidationIcon />;
  }
  return <UnsetValidationIcon />;
};

const ValidationMessage = ({ message, active }) => (
  <Row>
    <li
      className={`validation__message ${active === false ? 'validation__message--inactive' : ''}`}
    >
      <div className="validation__message__icon-container">
        <ValidationIcon active={active} />
      </div>
      <span className="validation__message__text">{message}</span>
    </li>
  </Row>
);

ValidationMessage.propTypes = {
  message: PropTypes.string,
  active: PropTypes.bool,
};
ValidationMessage.defaultProps = {
  message: undefined,
  active: undefined,
};

const ValidationMessages = ({ messages }) => {
  let validationMessageList = null;
  if (messages.length) {
    validationMessageList = messages.map(({ message, active }, i) => (
      <ValidationMessage key={i.toString()} message={message} active={active} />
    ));
  }
  return <ul className="validation__messages">{validationMessageList}</ul>;
};

ValidationMessages.propTypes = {
  messages: PropTypes.arrayOf(PropTypes.shape(ValidationMessage.propTypes)),
};

ValidationMessages.defaultProps = {
  messages: [],
};

const InvalidIcon = ({ size }) => (
  <Icon
    size={iconSizes[size]}
    name="warning-sign"
    className="mb-input__inner-icon validation__icon validation__icon--invalid "
    tooltip="Invalid"
    tooltipDelay={500}
    tooltipKind="error"
  />
);

const ValidationWrapper = ({ messages, active, children }) => (
  <Tooltip
    position="right"
    kind="neutral"
    custom
    delay={500}
    content={<ValidationMessages messages={messages} />}
    forceVisibility={active === true && messages && messages.length > 0}
    showOnHover={false}
  >
    {children}
  </Tooltip>
);

const NamedIcon = ({ namedIconTitle }) => <div className="named__icon">{namedIconTitle[0]}</div>;

NamedIcon.propTypes = {
  namedIconTitle: PropTypes.string,
};
NamedIcon.defaultProps = {
  namedIconTitle: undefined,
};

export { Loader, Placeholder, InnerButton, ValidationWrapper, InvalidIcon, NamedIcon };
