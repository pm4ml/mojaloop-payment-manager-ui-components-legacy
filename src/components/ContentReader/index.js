import ContentReader from './ContentReader';

export default ContentReader;
