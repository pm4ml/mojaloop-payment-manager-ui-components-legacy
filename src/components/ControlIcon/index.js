import ControlIcon from './ControlIcon';

export default ControlIcon;
