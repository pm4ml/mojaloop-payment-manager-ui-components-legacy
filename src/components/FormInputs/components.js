import './FormInput.scss';
import '../../icons/mule/forum-small.svg';

import React, { PureComponent } from 'react';

import * as utils from '../../utils/common';
import Button from '../Button';
import Icon from '../Icon';
import Tooltip from '../Tooltip';

export const FieldInfoOverlay = ({ assignRef, title, description, url, required }) => (
  <div className="forminput__field-info" ref={assignRef}>
    <div className="forminput__field-info__title-container">
      <div className="forminput__field-info__title">{title}</div>
      {required && (
        <React.Fragment>
          <div className="forminput__field-info__required-icon">
            <Icon size={20} name="info-small" className="forminput__label-icon" fill="#333" />
          </div>
          <div className="forminput__field-info__required-label"> required </div>
        </React.Fragment>
      )}
    </div>
    <div className="forminput__field-info__content">
      <p>{description}</p>
    </div>
    {url && (
      <div className="forminput__field-info__url">
        <a href={url}>{url}</a>
      </div>
    )}
  </div>
);

export class FieldInfo extends PureComponent {
  constructor() {
    super();
    this.onClick = this.onClick.bind(this);
    this.open = this.open.bind(this);
    this.close = this.close.bind(this);
    this.closeIfClickingOutside = this.closeIfClickingOutside.bind(this);
    this.state = {
      open: false,
    };
  }
  componentWillUnmount() {
    this.close();
  }
  onClick() {
    if (this.state.open) {
      this.close();
    } else {
      this.open();
    }
  }
  open() {
    this.area = React.createRef();
    this.setState(
      {
        open: true,
      },
      () => {
        document.addEventListener('click', this.closeIfClickingOutside);
      },
    );
  }
  close() {
    this.setState(
      {
        open: false,
      },
      () => {
        this.area = null;
        document.removeEventListener('click', this.closeIfClickingOutside);
      },
    );
  }
  closeIfClickingOutside(evt) {
    if (!this.area.current) {
      return;
    }
    if (!this.area.current.contains(evt.target)) {
      this.close();
    }
  }
  render() {
    const { iconSize, title, required, description, url } = this.props;
    let content = null;
    if (this.state.open) {
      content = (
        <FieldInfoOverlay
          assignRef={this.area}
          title={title}
          description={description}
          url={url}
          required={required}
        />
      );
    }
    const fieldInfoClassName = utils.composeClassNames([
      'forminput__field-info__icon',
      this.state.open && 'forminput__field-info__icon--active',
    ]);

    return (
      <Tooltip
        forceVisibility={this.state.open}
        showOnHover={false}
        content={content}
        custom
        position="right"
        align="start"
      >
        <Icon
          name="forum-small"
          onClick={this.onClick}
          size={iconSize}
          className={fieldInfoClassName}
        />
      </Tooltip>
    );
  }
}

export const Label = ({ size, label, required, complete, description, url }) => {
  if (!label) {
    return null;
  }

  const className = utils.composeClassNames([
    'forminput__label-box',
    size === 's' && 'forminput__label-box--small',
    size === 'm' && 'forminput__label-box--medium',
    size === 'l' && 'forminput__label-box--large',
  ]);
  const iconSizes = {
    s: 11,
    m: 12,
    l: 14,
  };
  const iconSize = iconSizes[size];
  return (
    <div className={className}>
      {required && (
        <Icon
          size={iconSize}
          name="info-small"
          className="forminput__label-icon"
          fill={complete ? '#39f' : '#f93'}
          tooltip={complete ? '' : 'This is a required field'}
        />
      )}
      <label>{label}</label>
      {description && (
        <FieldInfo
          iconSize={iconSize}
          title={label}
          description={description}
          url={url}
          required={required}
        />
      )}
    </div>
  );
};

export const InlineButton = ({ visible, isLocked, isDisabled, onClick, label }) => {
  if (!visible || isLocked || isDisabled) {
    return null;
  }

  return (
    <Button
      size="m"
      noFill
      kind="secondary"
      className="forminput__inline-button"
      onClick={onClick}
      disabled={isDisabled}
      label={label}
    />
  );
};

export const LockedIcon = ({ locked }) => {
  if (!locked) {
    return null;
  }
  return (
    <div className="forminput__inline-icon">
      <Icon
        name="lock-small"
        size={20}
        fill="#999"
        style={{ marginLeft: '10px' }}
        tooltip="this field is locked"
      />
    </div>
  );
};

export const InfoMessage = ({ message }) => (
  <div className="forminput__input-message">{message && <Tooltip>{message}</Tooltip>}</div>
);
