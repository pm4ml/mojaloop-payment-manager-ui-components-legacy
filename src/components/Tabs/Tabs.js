import './Tabs.scss';

import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

import * as utils from '../../utils/common';
import keyCodes from '../../utils/keyCodes';

class Tab extends PureComponent {
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }
  onClick(e) {
    if (!this.props.disabled && !this.props.selected) {
      this.props.onSelect(e);
    }
  }
  render() {
    const { selected, focused, children, disabled, hidden, flex, style } = this.props;
    if (hidden) {
      return null;
    }
    const className = utils.composeClassNames([
      'el',
      'el-tabs__tab-item',
      focused && 'el-tabs__tab-item--focused',
      selected && 'el-tabs__tab-item--selected',
      disabled && 'el-tabs__tab-item--disabled',
      flex && 'fill-width',
    ]);

    return (
      <div onClick={this.onClick} className={className} style={style} role="presentation">
        {children}
      </div>
    );
  }
}

Tab.propTypes = {
  children: PropTypes.node,
  selected: PropTypes.bool,
  focused: PropTypes.bool,
  onSelect: PropTypes.func,
  disabled: PropTypes.bool,
  hidden: PropTypes.bool,
  flex: PropTypes.bool,
  style: PropTypes.shape(),
};

Tab.defaultProps = {
  children: undefined,
  selected: false,
  focused: false,
  onSelect: undefined,
  disabled: false,
  hidden: false,
  flex: false,
  style: undefined,
};

const TabList = ({ children }) => children;

TabList.propTypes = {
  children: PropTypes.node,
};

TabList.defaultProps = {
  children: undefined,
};

const TabPanel = ({ children, flex, style, className }) => {
  const classNames = utils.composeClassNames([
    'el-tabs__tab__content',
    flex && 'el-tabs__tab__content--flexible',
    className,
  ]);
  return (
    <div className={classNames} style={style}>
      {children}
    </div>
  );
};

TabPanel.propTypes = {
  style: PropTypes.shape(),
  className: PropTypes.string,
  children: PropTypes.node,
  flex: PropTypes.bool,
};

TabPanel.defaultProps = {
  style: undefined,
  className: undefined,
  children: undefined,
  flex: false,
};

const TabPanels = ({ children }) => children;

TabPanels.propTypes = {
  children: PropTypes.node,
};

TabPanels.defaultProps = {
  children: undefined,
};

const TAB_TYPE = (<Tab />).type;
const TAB_PANEL_TYPE = (<TabPanel />).type;

const isTab = node => node.type === TAB_TYPE;
const isTabPanel = node => node.type === TAB_PANEL_TYPE;

class Tabs extends PureComponent {
  constructor(props) {
    super(props);

    const { selected, children } = this.props;
    const items = this.getTabs(children);
    let index = 0;
    if (typeof selected === 'string') {
      index = items.indexOf(selected);
    } else if (selected !== undefined) {
      index = selected;
    }

    this.state = {
      selected: Math.max(0, index),
      focused: undefined,
    };

    this.onSelect = this.onSelect.bind(this);
    this.onFocus = this.onFocus.bind(this);
    this.onBlur = this.onBlur.bind(this);

    this.getTabListAndTabPanels = this.getTabListAndTabPanels.bind(this);
    this.getTabs = this.getTabs.bind(this);
    this.getPanels = this.getPanels.bind(this);

    this.testKey = this.testKey.bind(this);
    this.selectSiblingTab = this.selectSiblingTab.bind(this);
  }
  componentDidUpdate(prevProps) {
    let currentSelected = this.state.selected;
    if (prevProps.selected !== this.props.selected) {
      currentSelected = this.props.selected;
    }

    const tabs = this.getTabs();
    const { hidden, disabled } = tabs[currentSelected].props;
    if (hidden || disabled) {
      currentSelected = 0;
    }

    this.setState({ selected: currentSelected });
  }

  onSelect(evt, index) {
    this.setState({ selected: index, focused: index });

    this.input.focus();

    if (this.props.onSelect) {
      this.props.onSelect(evt, index);
    }
  }
  onFocus(e) {
    e.preventDefault();
    e.stopPropagation();
    if (this.state.focused === undefined) {
      this.setState(state => ({ focused: state.selected }));
    }
  }
  onBlur() {
    this.setState({ focused: undefined });
  }

  getTabListAndTabPanels() {
    const items = this.props.children;
    const subElements = Array.isArray(items) ? items : [items];
    const [tabList, tabPanels] = subElements;
    return [tabList, tabPanels];
  }
  getTabs() {
    const [tabList] = this.getTabListAndTabPanels();
    const { children } = tabList.props;
    const tabs = Array.isArray(children) ? children : [children];
    return tabs.filter(isTab);
  }
  getPanels() {
    const tabPanels = this.getTabListAndTabPanels()[1];
    let panels = [];
    if (tabPanels) {
      const { children } = tabPanels.props;
      panels = Array.isArray(children) ? children : [children];
    }
    return panels.filter(isTabPanel);
  }
  testKey(e) {
    const { keyCode, shiftKey } = e.nativeEvent;
    if (keyCode === keyCodes.KEY_TAB) {
      e.preventDefault();
      this.setState({ focused: undefined });
      utils.focusNextFocusableElement(this.input, !shiftKey);
      return;
    }
    if (keyCode === keyCodes.KEY_LEFT) {
      e.preventDefault();
      this.selectSiblingTab(false);
      return;
    }
    if (keyCode === keyCodes.KEY_RIGHT) {
      e.preventDefault();
      this.selectSiblingTab(true);
    }
  }
  selectSiblingTab(next) {
    const { selected } = this.state;
    const tabs = this.getTabs(this.props.children);
    let nextIndex = selected;
    const found = false;
    while (!found) {
      nextIndex += next ? 1 : -1;
      if (nextIndex === tabs.length || nextIndex < 0) {
        break;
      }
      if (!tabs[nextIndex].props.disabled) {
        this.setState({ selected: nextIndex, focused: nextIndex });
        break;
      }
    }
  }
  render() {
    const { selected, focused } = this.state;
    const { id, children, flex } = this.props;
    const [tabList] = this.getTabListAndTabPanels();
    let shouldRenderAsFlex = flex;

    const { width } = tabList.props.style || {};
    const growTab = width !== undefined;
    const tabs = this.getTabs(children).map((child, index) => {
      const props = {
        ...child.props,
        onSelect: evt => this.onSelect(evt, index),
        key: index,
        selected: selected === index && !child.props.disabled,
        focused: focused === index,
        flex: growTab,
      };
      return React.cloneElement(child, props);
    });

    const panels = this.getPanels();
    let panel = null;
    if (panels.length >= selected + 1) {
      if (tabs[selected].props.selected) {
        panel = panels[selected];
        shouldRenderAsFlex = flex || panel.props.flex;
        panel = React.cloneElement(panel, { ...panel.props, flex: shouldRenderAsFlex });
      }
    }

    const tabsClassNames = utils.composeClassNames([
      'element',
      'el-tabs',
      shouldRenderAsFlex === true && 'el-tabs--flexible',
    ]);

    return (
      <div className={tabsClassNames} id={id}>
        <input
          ref={input => {
            this.input = input;
          }}
          type="button"
          className="el__holder"
          onFocus={this.onFocus}
          onBlur={this.onBlur}
          onKeyDown={this.testKey}
        />
        <div className="el-tabs__tab-items" style={tabList.props.style}>
          {tabs}
        </div>
        {panel}
      </div>
    );
  }
}

Tabs.propTypes = {
  id: PropTypes.string,
  selected: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  onSelect: PropTypes.func,
  disabled: PropTypes.bool,
  children: PropTypes.node,
  flex: PropTypes.bool,
};

Tabs.defaultProps = {
  id: 'el-tabs',
  selected: 0,
  onSelect: undefined,
  disabled: false,
  children: undefined,
  flex: false,
};

export { Tab, Tabs, TabList, TabPanel, TabPanels };
