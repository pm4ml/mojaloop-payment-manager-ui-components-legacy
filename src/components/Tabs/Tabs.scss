@import '../../assets/styles/vars/colors.scss';

.el-tabs {
}
.el-tabs--flexible {
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  min-height: 0;
}

.el-tabs__tab-items {
  display: flex;
  flex-grow: 1;
  flex-direction: row;
  padding-bottom: 10px;
  flex: 0 0 auto;
}
.el-tabs__tab-item {
  flex-grow: 0;
  flex-shrink: 1;
  flex-basis: auto;

  height: 32px;
  font-size: 14px;
  line-height: 32px;
  text-align: center;
  justify-content: flex-start;
  align-items: center;
  color: $color__dark-gray-3;
  border-bottom: 3px solid $color__gray-2;
  transition: all 0.1s ease-out;
  padding: 0px 10px;

  &.fill-width {
    flex-grow: 1;
  }

  &:hover:not(.el-tabs__tab-item--disabled):not(.el-tabs__tab-item--selected) {
    border-bottom-color: $color__gray-4;
    cursor: pointer;
  }
  &.el-tabs__tab-item--selected {
    border-bottom-color: $color__main-primary;
    color: $color__main-primary;
    cursor: pointer;
  }
  &.el-tabs__tab-item--focused {
    border-bottom-color: $color__main-primary--active;
    color: $color__main-primary--active;
  }
  &.el-tabs__tab-item--disabled {
    color: $color__main-disabled;
    cursor: default;
  }
}

.el-tabs__tab__content {
  &.el-tabs__tab__content--flexible {
    display: flex;
    flex-grow: 1;
    flex-direction: column;
    min-height: 0;
  }
}
.el-tabs__tab__panel {
  display: flex;
  flex-grow: 1;
  flex-direction: column;
}
