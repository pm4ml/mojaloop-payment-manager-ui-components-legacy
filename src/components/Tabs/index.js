import { Tab, TabList, TabPanel, TabPanels, Tabs } from './Tabs';

export { Tab, Tabs, TabList, TabPanel, TabPanels };
