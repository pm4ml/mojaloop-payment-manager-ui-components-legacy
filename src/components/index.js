import '../assets/styles/index.scss';

import Button from './Button';
import Checkbox from './Checkbox';
import Column from './Column';
import ContentReader from './ContentReader';
import ControlIcon from './ControlIcon';
import DataList, { Link } from './DataList';
import DatePicker from './DatePicker';
import Dropdown from './Dropdown';
import FileUploader from './FileUploader';
import FormInput, { FormInputs, FormInputsGroup, Inline } from './FormInputs';
import Heading from './Heading';
import Icon from './Icon';
import Menu, { MenuItem, MenuSection } from './Menu';
import MessageBox from './MessageBox';
import Modal, { ModalTabsLayout } from './Modal';
import Navbar from './Navbar';
import Pill from './Pill';
import RadioGroup from './RadioGroup';
import Row from './Row';
import ScrollBox from './ScrollBox';
import Select from './Select';
import Spinner from './Spinner';
import { Tab, TabList, TabPanel, TabPanels, Tabs } from './Tabs';
import TextArea from './TextArea';
import TextField from './TextField';
import Toast from './Toast';
import Toggle from './Toggle';
import Tooltip from './Tooltip';

export {
  Button,
  Column,
  ContentReader,
  ControlIcon,
  Checkbox,
  DataList,
  DatePicker,
  Dropdown,
  FileUploader,
  FormInput,
  FormInputs,
  FormInputsGroup,
  Heading,
  Icon,
  Inline,
  Link,
  Menu,
  MenuSection,
  MenuItem,
  MessageBox,
  Modal,
  ModalTabsLayout,
  Navbar,
  Pill,
  RadioGroup,
  Row,
  ScrollBox,
  Select,
  Spinner,
  Tabs,
  Tab,
  TabList,
  TabPanels,
  TabPanel,
  TextArea,
  TextField,
  Toast,
  Toggle,
  Tooltip,
};
