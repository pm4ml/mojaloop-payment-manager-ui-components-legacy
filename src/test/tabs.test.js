import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

import { Tab, TabList, TabPanel, TabPanels, Tabs } from '../components/Tabs';

const TabComponents = (
  <Tabs id="test-tabs" flex>
    <TabList>
      <Tab>Tab1</Tab>
      <Tab>Tab2</Tab>
      <Tab>Tab3</Tab>
      <Tab disabled>Tab4</Tab>
    </TabList>
    <TabPanels>
      <TabPanel>TabPanel1</TabPanel>
      <TabPanel>TabPanel2</TabPanel>
      <TabPanel>TabPanel3</TabPanel>
      <TabPanel>TabPanel4</TabPanel>
    </TabPanels>
  </Tabs>
);

it('renders the tabs', () => {
  const wrapper = mount(TabComponents);
  expect(wrapper.find('.el-tabs')).toBeTruthy();
});

it('renderd the prop id', () => {
  const wrapper = mount(TabComponents);
  expect(wrapper.find('#test-tabs')).toBeTruthy();
});

it('renderd the prop flex', () => {
  const wrapper = mount(TabComponents);
  expect(wrapper.find('.el-tabs--flexible')).toBeTruthy();
});

it('renders all the tab items', () => {
  const wrapper = mount(TabComponents);
  expect(wrapper.find(Tab)).toHaveLength(4);
});

it('renders only 1 tabpanel', () => {
  const wrapper = mount(TabComponents);
  expect(wrapper.find(TabPanel)).toHaveLength(1);
});

it('renders the first tabpanel content', () => {
  const wrapper = mount(TabComponents);
  expect(wrapper.find(TabPanel).text()).toBe('TabPanel1');
});

it('renders the second tabpanel content when second tab is clicked', () => {
  const wrapper = mount(TabComponents);
  wrapper
    .find(Tab)
    .at(1)
    .simulate('click');
  expect(wrapper.find(TabPanel).text()).toBe('TabPanel2');
});

it('does not renders the tabpanel of a disabled tab when clicked', () => {
  const wrapper = mount(TabComponents);
  wrapper
    .find(Tab)
    .at(3)
    .simulate('click');
  expect(wrapper.find(TabPanel).text()).not.toBe('TabPanel4');
});

it('does not renders the tabpanel when disabled by default disabled', () => {
  const wrapper = mount(
    <Tabs id="test-tabs" flex>
      <TabList>
        <Tab disabled>Tab1</Tab>
      </TabList>
      <TabPanels>
        <TabPanel>TabPanel1</TabPanel>
      </TabPanels>
    </Tabs>,
  );
  expect(wrapper.find(TabPanel).exists()).toBe(false);
});

it('does not renders the tab as selected when it is the only one and disabled', () => {
  const wrapper = mount(
    <Tabs id="test-tabs" flex>
      <TabList>
        <Tab disabled>Tab1</Tab>
      </TabList>
    </Tabs>,
  );
  expect(wrapper.find(Tab).exists()).toBe(true);
  expect(wrapper.find(Tab).prop('disabled')).toBe(true);
  expect(wrapper.find(Tab).prop('selected')).toBe(false);
});

it('triggers onSelect when clicking a different tab', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(
    <Tabs onSelect={mockEvent}>
      <TabList>
        <Tab>Tab1</Tab>
        <Tab>Tab2</Tab>
      </TabList>
    </Tabs>,
  );
  wrapper
    .find(Tab)
    .at(1)
    .simulate('click');
  expect(mockEvent).toHaveBeenCalled();
});

it('does not trigger onSelect when clicking a disabled tab', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(
    <Tabs onSelect={mockEvent}>
      <TabList>
        <Tab>Tab1</Tab>
        <Tab disabled>Tab2</Tab>
      </TabList>
    </Tabs>,
  );
  wrapper
    .find(Tab)
    .at(1)
    .simulate('click');
  expect(mockEvent).not.toHaveBeenCalled();
});

it('does not trigger onSelect when clicking the currently selected tab', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(
    <Tabs onSelect={mockEvent}>
      <TabList>
        <Tab>Tab1</Tab>
        <Tab>Tab2</Tab>
      </TabList>
    </Tabs>,
  );
  wrapper
    .find(Tab)
    .at(0)
    .simulate('click');
  expect(mockEvent).not.toHaveBeenCalled();
});

// Snapshot

it('renders the checkbox correctly when multiple props are set', () => {
  const wrapper = shallow(TabComponents);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
