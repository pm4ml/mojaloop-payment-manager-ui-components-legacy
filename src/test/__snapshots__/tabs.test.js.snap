// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the checkbox correctly when multiple props are set 1`] = `
<div
  className="element el-tabs el-tabs--flexible"
  id="test-tabs"
>
  <input
    className="el__holder"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    type="button"
  />
  <div
    className="el-tabs__tab-items"
  >
    <Tab
      disabled={false}
      flex={false}
      focused={false}
      hidden={false}
      key="0"
      onSelect={[Function]}
      selected={true}
    >
      Tab1
    </Tab>
    <Tab
      disabled={false}
      flex={false}
      focused={false}
      hidden={false}
      key="1"
      onSelect={[Function]}
      selected={false}
    >
      Tab2
    </Tab>
    <Tab
      disabled={false}
      flex={false}
      focused={false}
      hidden={false}
      key="2"
      onSelect={[Function]}
      selected={false}
    >
      Tab3
    </Tab>
    <Tab
      disabled={true}
      flex={false}
      focused={false}
      hidden={false}
      key="3"
      onSelect={[Function]}
      selected={false}
    >
      Tab4
    </Tab>
  </div>
  <TabPanel
    flex={true}
  >
    TabPanel1
  </TabPanel>
</div>
`;
